### PR TITLE
JAMES-1726 Update can introduce duplicate

### PR DESCRIPTION
**Pull request by Laura-Royet**
Add a test showing that update can create duplicate.
Given the following tree :
A / B / C A / C
Moving C (child of B) to A introduce a name duplication: this should not be permitted.

### CODE REVIEW

**Comment by rouazana** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMailboxesMethodTest.java [hunk 1]
```diff
@@ -1512,4 +1512,47 @@ public void setMailboxesShouldReturnNotUpdatedWhenRenameToSystemMailboxName() {
                     hasEntry(equalTo("type"), equalTo("invalidArguments")),
                     hasEntry(equalTo("description"), equalTo("The mailbox 'outbox' is a system mailbox.")))));
     }
+
+    @Test
+    public void setMailboxesShouldReturnNotUpdatedWhenMovingAMailboxInAParentAlreadyContainingIt() {
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMailboxesMethodTest.java]: setMailboxesShouldReturnNotUpdatedErrorWhenMovingMailboxTriggersNameConflict

**Comment by aduprat** [comment]
:+1:

**Comment by rouazana** [comment]
:+1:

**Comment by mbaechler** [comment]
merged
